### PR TITLE
Lower star rating pp exponent and increase accuracy value on high SR maps

### PR DESCRIPTION
**Pull request by MillhioreF**
As proposed [here](https://docs.google.com/document/d/1RJfGKcXfiiA7uMaIROzY3VF9mgexAgVsS1Ls2ADTojM/edit#).

tl;dr
Lower exponential growth on StrainBase to deflate pp gains as star rating goes up
StrainMultiplier modification to rebalance values of different score ranges
Involve strain in accuracy pp to increase it for 960K+ scores

### CODE REVIEW

**Comment by PrismaticEffect** [comment]
This is Shoegazer; thank you very much for implementing this. The changes perfectly align with what is written in the document.

I will be fiddling around with the AccValue exponent to avoid 960-985K scores being too overpowered for certain SRs compared to 800-850K scores; I think that 960K-985K give a touch bit too much AccValue bonus as it is, but I need to find a proper exponent first.

**Comment by WalterToro** [comment]
After reprocessing the mania scores, here is the Info from the top 2000 players: https://goo.gl/S1mFVy
Need to note that players who are inactive cant be reprocessed so their pp is noted as 0.

Notable things:
- Jakads (AscensionDTWhen) loses 10300,9 pp (the only regular player losing over 10k pp)
- Only one regular player wins pp, Asrielly (130,7 pp), that is a good acc player without DT plays
- Top mania pp play drops from 2414pp to 1482,43pp

In conclusion, the proposal here seems to fulfill its objective and its one step forward into fixing and scaling the mania pp.

**Comment by peppy** [comment]
Looking better, but i think we should still add a post-scale to get the highest pp total down to the same level at osu!.

**Comment by MillhioreF** [comment]
There was already one in the code so I've modified it now. The #1 player would drop down to 15,300pp with this change with the rest scaled accordingly.

**Comment by PrismaticEffect** [comment]
Not that the specific linear multiplier would mean much (since a linear multiplier doesn't do much other than do scaling), but a multiplier of 0.75x would make it align closer to osu!standard. I feel that using the #1 osu!mania player for calibration is ill-advised since he is so far ahead of everyone else (7 out of 10 of o!m's highest PP plays are his with the new system). I would say that using the top 5 players in general to calibrate PP scaling would be a stretch, for those players are notably far ahead of everyone else.  

There will be about 95 players would have 10,000+ pp with a 0.75x multiplier. In contrast, only 27 players will have 10,000+ pp with a 0.65x player. osu!standard currently has 188 players with 10,000+ pp, though the amount of players can be slightly lower with the HD/short map changes.

You could even use a 0.8x multiplier since there are so many osu!standard players with 10,000pp since the main goal is to adjust the PP values to align with osu!standard. About 150 players would have 10,000+ pp on osu!mania with a 0.8x multiplier, which is slightly lower than the amount of osu!standard players with 10,000+ pp or higher.

**Comment by MillhioreF** [comment]
I'm on board with 0.8. Crunching pp values to be sane is fair, but I don't want to lower it too much either - scores are already extremely homogenous at beginner levels of play and changing the multiplier is only going to make them more so.

**Comment by PrismaticEffect** [comment]
The homogeneity was a concern of mine as well, thank you for the change. One last thing because Walter and I overlooked -- the multiplier should be 0.88x. There was a previous multiplier in the original PP system of 1.1, which means that changing the multiplier from 1.1x to 0.8x would reduce the multiplier by 27.7% instead of 20%. Our mistake for overlooking that.

**Comment by WalterToro** [comment]
After discussing with @PrismaticEffect we came into an agreement that the 0.8x multiplier is good, looking at the possible new ranked maps in mania and the pp those can give in the long run.
With that, this proposal seems to be ready for merge.

**Comment by Tom94** [comment]
23k pp is still over 1.5x larger than 14k, which is the highest in standard. I think if we go with reducing the overall multiplier, then we should do it over-convervatively (i.e. aim to be below standard) rather than under-conservatively. The trend in mania is for pp to increase faster than standard due to more crazy maps getting ranked, so I believe this will all equalize out.

**Comment by WalterToro** [comment]
Yep, the 23k pp was still a big number, so we proceeded to change the scaling multiplier that was on the code (originally 1.1x, we dont know why), millhiore originally put 0.65x but we proposed 0.8x to be akin to the other leaderboards (more exactly the ctb leaderboard).
The results should be close to the numbers on the strong yellow column here https://goo.gl/YJ7S3a (tdlr, the drop will be from 33.4k to around 16.9k pp for the top player, and around 72 active players will get over 10k pp with this new scale)

**Comment by Tom94** [comment]
I see, looks like I misunderstood the sheet. Let's go then. :)

**Comment by nl-tatatat** [comment]
But why? I love the big pp, and I wish all gamemodes had big pp like mania. I want 1000pp plays, not 400pp plays.

**Comment by tailsdk** [comment]
I think this change is pretty good i dont think the pp reduction was needed as in the multiplier but i can see the accuracy focus being a good change. I can also see why peppy wants to lower the pp to a level like the other gamemodes since it would make sense as a develeper to keep some sort of balance between all the gamemodes.

**Comment by peppy** [comment]
The previous values were overflowing database/network values. It was unintended.
